### PR TITLE
fix crash when profiler not enabled

### PR DESCRIPTION
**Pull request by cjolivier01**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/engine/threaded_engine.h [hunk 1]
```diff
@@ -398,7 +398,8 @@ class ThreadedEngine : public Engine {
   }
 
   int bulk_size() const override {
-    return profiler::Profiler::Get()->AggregateRunning() ? 0 :  BulkStatusStore::Get()->bulk_size;
+    const profiler::Profiler *prof = profiler::Profiler::Get();
```
Comment (piiswrong) [inline src/engine/threaded_engine.h]: why would profiler::get return nullptr when it's not enabled?

**Comment by piiswrong** [comment]
@marcoabreu any idea why CI wasn't able to catch this?

**Comment by szha** [comment]
Is there a build that doesn't turn on profiler?

File: src/engine/threaded_engine.h (hunk 1 above)
Comment (cjolivier01) [reply src/engine/threaded_engine.h]: I didn't think it did either. When profiler isn't enabled in the build at all.
The ability to turn profiler off in the build will be removed soon

**Comment by cjolivier01** [comment]
I think I'll just remove the ability to turn off profiler at build time tomorrow

**Comment by cjolivier01** [comment]
Here is removing: https://github.com/apache/incubator-mxnet/pull/10308
With that change, this PR isn't needed

**Comment by cjolivier01** [comment]
everyone ok for this to merge when it completes?

**Comment by sxjscience** [comment]
I'm ok for merging.

**Comment by cjolivier01** [comment]
CI is starting to remind me of the "old days" when it'd take days to get a build to go through without a strange failure...  This is like build attempt #5 or something.

**Comment by marcoabreu** [comment]
@piiswrong I think all builds in CI are turning on profiler by default

**Comment by cjolivier01** [comment]
attempt number 6 to build with CI...

**Comment by marcoabreu** [comment]
I know the notify_shutdown() is only an attempt to fix the hanging gpu test, but could you make sure to remove it before you merge this PR?

**Comment by cjolivier01** [comment]
1) Nope
2) It was just an experimental  change to see if it helped, so it will be
reverted

**Pull request merged**